File: POOVeiculos/Python/Moto.py
from Veiculo import Veiculo

class Moto(Veiculo):
    def engatarRe(self, r):
        if r.upper() == "R":
            print("Moto não tem ré! Empurre a moto!")
        else:
            print("Não é possível engatar essa marcha no {} {}.".format(self.__class__.__name__.lower(), self.__modelo))

    def __init__(self, modelo, tipo, potencia, ano):
        super().__init__(modelo, tipo, potencia, ano)

    def __del__(self):
        print("Objeto Moto Destruído com Sucesso!")

File: POOVeiculos/C++/Moto.cpp
#include "Moto.h"

void Moto::engatar(char r) {
    if((char)toupper(r) == 'R') {
        cout << "Moto não tem ré! Empurre a moto!" << endl;
    }
    else {
        cout << "Não é possível engatar essa marcha no " << typeid(*this).name() << " " << this->modelo << "." << endl;
    }
}

Moto::Moto(string modelo, string tipo, float potencia, int ano) : Veiculo(modelo, tipo, potencia, ano) {

}

Moto::~Moto() {
    cout << "Objeto Moto Destruído com Sucesso!" << endl;
}
